src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets

```diff
@@ -5,29 +5,54 @@
     <Message Text="Running Serialization Tool" Importance="normal" />
-    <Exec Command="dotnet Microsoft.XmlSerializer.Generator &quot;$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)&quot; --force --quiet --reference &quot;@(ReferencePath)&quot;" ContinueOnError="true"/>
+    <Exec Command="dotnet Microsoft.XmlSerializer.Generator &quot;$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)&quot; --force --quiet $(_SgenRspFilePath)" ContinueOnError="true"/>
     <Warning Condition="Exists('$(_SerializerCsIntermediateFolder)') != 'true'" Text="$(_SGenWarningText)" />
-    <Csc Condition="Exists('$(_SerializerCsIntermediateFolder)') == 'true'" ContinueOnError="true" OutputAssembly="$(_SerializerDllIntermediateFolder)" References="@(ReferencePath);@(IntermediateAssembly)" EmitDebugInformation="$(DebugSymbols)" Sources="$(_SerializerCsIntermediateFolder)" TargetType="Library" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" DisabledWarnings="$(_SerializationAssemblyDisabledWarnings)"/>
+    <Csc Condition="Exists('$(_SerializerCsIntermediateFolder)') AND !Exists('$(_CscRspFilePath)')" ContinueOnError="true" OutputAssembly="$(_SerializerDllIntermediateFolder)" References="@(ReferencePath);@(IntermediateAssembly)" EmitDebugInformation="$(DebugSymbols)" Sources="$(_SerializerCsIntermediateFolder)" TargetType="Library" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" DisabledWarnings="$(_SerializationAssemblyDisabledWarnings)"/>
+    <Csc Condition="Exists('$(_SerializerCsIntermediateFolder)') AND Exists('$(_CscRspFilePath)')" ContinueOnError="true" OutputAssembly="$(_SerializerDllIntermediateFolder)" References="@(ReferencePath);@(IntermediateAssembly)" EmitDebugInformation="$(DebugSymbols)" Sources="$(_SerializerCsIntermediateFolder)" TargetType="Library" ResponseFiles="$(_CscRspFilePath)"  ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" DisabledWarnings="$(_SerializationAssemblyDisabledWarnings)"/>
     <Warning Condition="Exists('$(_SerializerDllIntermediateFolder)') != 'true' And Exists('$(_SerializerCsIntermediateFolder)') == 'true'" Text="$(_SGenWarningText)"/>
     <Copy Condition="Exists('$(_SerializerDllIntermediateFolder)') == 'true'" SourceFiles="$(_SerializerDllIntermediateFolder)" DestinationFolder="$(OutputPath)" />
+    <Delete Condition="Exists('$(_SgenRspFilePath)')" Files="$(_SgenRspFilePath)" />
+    <Delete Condition="Exists('$(_CscRspFilePath)')" Files="$(_CscRspFilePath)" />
+  </Target>
+
+  <Target Name="GenerateRspFiles" BeforeTargets="GenerateSerializationAssembly">
+    <PropertyGroup>
+      <SGenReferences Condition=" '$(SGenReferences)' == ''">@(ReferencePath)</SGenReferences>
+    </PropertyGroup>
+    <ItemGroup>
+      <SgenRspFile Include ="--reference $(SGenReferences.Replace(';', '%3B'))" />
+      <SgenRspFile Include ="--type $(SGenTypes.Replace(';', '%3B'))" Condition=" '$(SGenTypes)' != ''" />
+      <SgenRspFile Include ="--verbose" Condition=" '$(SGenVerbose)' == 'true'" />
```

Comment:
Why are you replacing ; with the hex value for it?

Reply:
@mconnew When add multiple values to an item, they are separated by **;** by default. 
(ex: --reference RefA;RefB**;**--verbose**;**--proxytypes). 
When WriteLinesToFile task executes, it will first look for all ; in the text and then replace them with '%0A', if there are ; in the value like RefA;RefB, WriteLinesToFile will take it as a separator and replace it with '%0A'.

Reply:
Interesting.